Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -144,21 +144,52 @@ public sealed class MemoryView : ICodeFormattable {
             }
         }
 
-        public object suboffsets {
+        public PythonTuple suboffsets {
             get {
                 CheckBuffer();
-                return _buffer.SubOffsets;
+                return _buffer.SubOffsets ?? PythonTuple.EMPTY;
             }
         }
 
         public Bytes tobytes() {
             CheckBuffer();
-            return _buffer.ToBytes(_start, _end);
+            if (_matchesBuffer && _step == 1) {
+                return _buffer.ToBytes(_start / _itemsize, _end / _itemsize);
+            }
+
+            byte[] bytes = getByteRange(_start, numberOfElements() * _itemsize);
+
+            if (_step == 1) {
+                return Bytes.Make(bytes);
+            }
+
+            // getByteRange() doesn't care about our _step, so if we have one
+            // that isn't 1, we will need to get rid of any bytes we don't care
+            // about and potentially adjust for a reversed memoryview.
+            byte[] stridedBytes = new byte[bytes.Length / _itemsize];
+            for (int indexStrided = 0; indexStrided < stridedBytes.Length; indexStrided += _itemsize) {
+
+                int indexInBytes = indexStrided * _step;
+
+                for (int j = 0; j < _itemsize; j++) {
+                    stridedBytes[indexStrided + j] = bytes[indexInBytes + j];
+                }
+            }
+
+            return Bytes.Make(stridedBytes);
         }
 
         public PythonList tolist() {
             CheckBuffer();
-            return _buffer.ToList(_start, _end);
```

Comment:
Is this a spot where we could have a fast path with `_matchesBuffer && _step == 1`?

Reply:
Added, but only tested against the two `test_memoryview` tests.